utils/terraform_client.py

```diff
@@ -0,0 +1,143 @@
+import sys
+import shutil
+import base64
+import logging
+
+from utils.openshift_resource import OpenshiftResource
+
+from python_terraform import Terraform
+
+
+class ConstructResourceError(Exception):
+    def __init__(self, msg):
+        super(ConstructResourceError, self).__init__(
+            "error construction openshift resource: " + str(msg)
+        )
+
+
+class OR(OpenshiftResource):
+    def __init__(self, body, integration, integration_version):
+        super(OR, self).__init__(
+            body, integration, integration_version
+        )
+
+
+class TerraformClient(object):
+    def __init__(self, integration, integration_version,
+                 integration_prefix, working_dirs):
+        self.integration = integration
+        self.integration_version = integration_version
+        self.integration_prefix = integration_prefix
+        self.working_dirs = working_dirs
+        tfs = {}
+        for name, wd in working_dirs.items():
+            tf = Terraform(working_dir=wd)
+            self.return_code, self.stdout, self.stderr = tf.init()
+            self.check_output()
```

Comment:
since we are looking at running these in parallel in the near future, i think it would be best to pass the return_code, stdout and stderr as arguments to check_output instead of storing them in self.

we run check_output after every terraform command (init and plan for example) to check their output. we can not run these command for all tf instances together, since they refer to different aws accounts, state buckets, credentials, etc. we must run them separately for each tf instance.

to be able to understand which tf instance failed, i added the tf name as an argument.

Reply:
that makes sense, thanks!